src/kvirc/ui/KviUserListView.cpp

```diff
@@ -485,60 +487,115 @@ void KviUserListView::completeNickBashLike(const QString & szBegin, std::vector<
 	}
 }
 
-bool KviUserListView::completeNickStandard(const QString & szBegin, const QString & szSkipAfter, QString & szBuffer, bool bAppendMask)
+bool KviUserListView::completeNickStandard(const QString & szBegin, QString & szSkipAfter, QString & szBuffer, bool bAppendMask)
 {
-	KviUserListEntry * pEntry = m_pHeadItem;
+	// For normal nickname completion, we'll build a list of matching nicknames in order,
+	// then look through it to the next nickname each time Tab is pressed.
+	if (m_CompletionList == nullptr)
+	{
+		// No list; build it.
+		m_CompletionList = new std::list<QString>();
+
+		KviUserListEntry * pEntry = m_pHeadItem;
+
+		for(KviUserListEntry * pEntry = m_pHeadItem; pEntry; pEntry = pEntry->m_pNext)
+		{
+			// FIXME: completion should skip my own nick or place it as last entry in the chain (?)
+
+			//	if(KviConsoleWindow * c = m_pKviWindow->console())
+			//	{
+			//		if(kvi_strEqualCI(entry->m_szNick.ptr(),c->currentNickName())
+			//	}
+
+			if(pEntry->m_szNick.length() >= szBegin.length())
```

Comment:
Consider using this instead and dropping the indention level of the rest of the code in the for statement.
`if(pEntry->m_szNick.length() < szBegin.length()) continue;`